Unity-Technologies/com.unity.multiplayer.samples.coop
# feat: flag to prevent dev and release builds to connect

### Description (*)

This PR adds a flag to the connection payload telling if the game instance is a dev build or not. The connection approval then uses it to deny connections with different values for that flag. Then, they get an error message telling them they are using incompatible build types.
This PR also simplifies and clarifies the connection approval flow.
### Related Pull Requests

### Issue Number(s) (*)

Fixes issue(s): [MTT-2521](https://jira.unity3d.com/browse/MTT-2521)
### Manual testing scenarios

1. Start a host in editor or in a dev build
2. Try to join as a client in a release build
3. See the error
4. The reverse scenario should also produce the same error
### Questions or comments

### Contribution checklist
